core/src/main/scala/org/bykn/bosatsu/TypedExprNormalization.scala

```diff
@@ -309,6 +310,35 @@ object TypedExprNormalization {
         val tpe = Type.normalize(tpe0)
         if (tpe == tpe0) None
         else Some(Local(n, tpe, tag))
+      case App(
```

Comment:
really a clean way to go will be to match on `IntAlgebraic` or something extractor that matches if we have an expression of an integer in some algebraic combination (add, sub, times, div) and then normalize that expression.

for instance `(x + 1) + 2` should be normalized to `x + 3` but this rule won't do it because it doesn't understand associativity (or commutativity).

Reply:
This was addressed.